### PR TITLE
add lint avoid_returning_null_for_void

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

**Comment by bwilkerson** [comment]
Has this been discussed somewhere? I don't remember the motivation (which should be added to the documentation) and would like to know it for context.

**Comment by a14n** [comment]
>Has this been discussed somewhere?

No

>I don't remember the motivation (which should be added to the documentation) and would like to know it for context.

Basically `return null` was used when `Null` was used instead of `void`. I find it clearer to used `return;` instead of `return null;` in a function with a return type `void`.

File: lib/src/rules/avoid_returning_null_for_void.dart [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r"Don't return null for void.";
+
+const _details = r'''
+
+**Don't** return null for void
```
Comment (pq) [inline lib/src/rules/avoid_returning_null_for_void.dart]: maybe replace w/ 

    **AVOID** returning null for void.

(I think that's more consistent on a quick scan.)

Comment (pq) [reply lib/src/rules/avoid_returning_null_for_void.dart]: Also, it would be great to have a sentence or two of rationale...

**Comment by a14n** [comment]
PTAL

File: lib/src/rules/avoid_returning_null_for_void.dart [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+```
+void f1() {
+  return;
+}
+Future<void> f2() async {
+  return;
+}
+```
+
+''';
```
Comment (pedromassango) [inline lib/src/rules/avoid_returning_null_for_void.dart:42]: What is the difference of the "GOOD" and "BAD" statements here?

**Pull request merged**